elixir/apps/web/lib/web/live/gateway_groups/new.ex

```diff
@@ -128,4 +124,35 @@ defmodule Web.GatewayGroups.New do
 
     {:noreply, socket}
   end
+
+  defp docker_command(secret) do
+    """
+    docker run -d \\
+      --name=firezone-gateway-0 \\
+      --restart=always \\
+      -v /dev/net/tun:/dev/net/tun \\
+      -e FZ_SECRET=#{secret} \\
+      us-east1-docker.pkg.dev/firezone/firezone/gateway:stable
+    """
+  end
+
+  defp systemd_command(_secret) do
+    """
+    [Unit]
+    Description=zigbee2mqtt
+    After=network.target
+
+    [Service]
+    ExecStart=/usr/bin/npm start
+    WorkingDirectory=/opt/zigbee2mqtt
+    StandardOutput=inherit
+    StandardError=inherit
+    Restart=always
+    User=pi
```

Comment:
```
[Unit]
Description=Start a Firezone Gateway container

[Service]
TimeoutStartSec=0
Restart=always
ExecStartPre=/usr/bin/docker pull ${container_image}
ExecStart=/usr/bin/docker run --rm --name=${container_name} --health-cmd="ip link | grep tun-firezone" --name=firezone-gateway --cap-add=NET_ADMIN --sysctl net.ipv4.ip_forward=1 --sysctl net.ipv4.conf.all.src_valid_mark=1 --sysctl net.ipv6.conf.all.disable_ipv6=0 --sysctl net.ipv6.conf.all.forwarding=1 --sysctl net.ipv6.conf.default.forwarding=1 --device="/dev/net/tun:/dev/net/tun" --env TOKEN="..." --env RUST_LOG="warn" --env ENABLE_MASQUERADE=1 ghcr.io/firezone/gateway:${FIREZONE_VERSION:-1}
ExecStop=/usr/bin/docker stop gateway
ExecStopPost=/usr/bin/docker rm gateway
```

Reply:
@AndrewDryga I think the point of Systemd here is for users who don't want / can't install Docker and just want to run the static binary

Reply:
@jamilbk let's just use this one for now, until we have a better one. Because right now it doesn't even deploy Firezone :)

Reply:
Should we just get rid of the systemd command tab altogether for the time being?

Reply:
@bmanifold Yeah I like that. It saves us from users end up using this one and us having to support two systemd approaches 🙃

Reply:
This one should do the job :)

```
[Unit]
Description=Firezone Gateway
After=network.target

[Service]
Type=simple
Environment="TOKEN=#{token}"
Environment="VERSION=#{token}"
ExecStartPre=/bin/sh -c ' \
  arch=$(uname -m); \
  case $$arch in \
    aarch64) \
      bin_url="https://github.com/firezone/firezone/releases/download/1.20231001.0/gateway-aarch64-unknown-linux-musl-1.20231001.0" ;; \
    armv7l) \
      bin_url="https://github.com/firezone/firezone/releases/download/1.20231001.0/gateway-armv7-unknown-linux-musleabihf-1.20231001.0" ;; \
    x86_64) \
      bin_url="https://github.com/firezone/firezone/releases/download/1.20231001.0/gateway-x86_64-unknown-linux-musl-1.20231001.0" ;; \
    *) \
      echo "Unsupported architecture"; \
      exit 1 ;; \
  esac; \
  wget -O /usr/local/bin/firezone-gateway $$bin_url \
'
ExecStartPre=/usr/bin/chmod +x /usr/local/bin/firezone-gateway
ExecStart=/usr/local/bin/firezone-gateway --token $TOKEN
Restart=always
RestartSec=3

[Install]
WantedBy=multi-user.target
```

Reply:
ChatGPT made instructions:


1. Create the systemd service file:

Open your favorite editor (e.g., nano, vim) with root privileges:

```
sudo nano /etc/systemd/system/firezone-gateway.service
```

2. Paste the service file contents:

Copy the provided systemd service unit content and paste it into the opened editor.

3. Save and close the editor:

If using nano, press CTRL + O to save and CTRL + X to exit. If using vim, press :wq and hit Enter.

4. Reload the systemd daemon:

This ensures that systemd recognizes your new service:
```
sudo systemctl daemon-reload
```

5. Enable and start the service:

Enable it so that it starts on boot:
```
sudo systemctl enable firezone-gateway
```

Start the service:
```
sudo systemctl start firezone-gateway
```

6. Ensure that the service has started successfully:

```
sudo systemctl status firezone-gateway
```

Reply:
BTW @jamilbk how are we going to show download links in the UI? Use the current portal version to download binaries of the same version? That might not work because not every release of a portal will lead to a release of all the binaries (eg. hotfixes) and our version scheme (that uses date) will make sure that the minor component changes daily.

For Docker I'm pushing a major version tag `1`, so that we can fetch those and auto-update. This will not work with binary download links by default.

Reply:
@AndrewDryga

- Artifacts attached to releases can be accessed with `latest` alias, so we can just link to that from the UI
- The version minor is our API version and doesn't change automatically. It only changes when we want it to by updating Makefile and running `make version`. The date is the API version date, similar to Stripe and GitHub.
- Based on the above, links to Docker images should be specified up to the minor so that they're API-compatible, e.g. `1.20231001`

Reply:
> Based on the above, links to Docker images should be specified up to the minor so that they remain API-compatible, e.g. 1.20231001

This means "never auto-upgrade" which is also an option, but want to make sure it's intentional. Because the very next version will be `1.20231002` and docker will not pull that on restart. 

My logic is the following: if you have v1 running and you restart a container and docker sees there is a more recent image (without major breaking changes) - it should pull the latest image for v1 (tagged `1`) and auto-upgrade. Because if that version is published to the GitHub Registry it also means our production is ready to accept connections from it.

Reply:
@AndrewDryga I think there might still be some confusion with how versioning is set up to work:

- Version `1.20231001.0` is our current version
- Version `1.20231001.1` is the next version and so on
- Version `1.20240401.0` would be the next minor version (for example) -- this would correspond to an API release date of 2024-04-01 -- 6 months later. This would break backwards compatibility. The portal could keep all clients / relays / gateways on the same API grouped together. We could eventually roll out an SLA to support, e.g. 2 API versions at a time for example (1 year)

We could use the API version as the major, e.g. `20240401.0.0` and then update it every 6 months as well. But then we really wouldn't use the patch version for anything, and it might be nice to keep the `1` as a marketing version, so we can roll out `Firezone 2.0` etc.

This is what was discussed in this thread: https://github.com/firezone/firezone/pull/2285#discussion_r1353108369

Reply:
Ah, got it! My *incorrect* assumption was that we would change the date part (`20231001`) every time `make version` is run, which means on every single release.

Reply:
Any concerns that using the US date format might confuse international customers? On a separate note, will gateways & clients both auto-update?

Reply:
@jefferenced I think we do want auto-updates, but we don't have such a feature right now. 

When Firezone is installed as a Docker container it will pull the latest version on restarts if we add `--pull always`, but we need to build a different mechanism when firezone is installed as a binary .

Reply:
@AndrewDryga This may be a silly suggestion, but can we force the gateways to restart periodically? And in doing so, pull the latest Docker container?

Reply:
We can do it for Docker, but it will have downsides in terms of availability. Would be better to have the portal push a message to one of them that they are outdated (it's a very simple feature for us because when a new portal version is deployed they all will have to reconnect and we receive client/relay/gateway versions when they are connecting).

The issue is then that we need some code that will download the new version, replace the existing container/binary, and restart it. This requires us to write some code for non-docker environments, for docker environments it's easier to automate but has downsides - we will pull the updates all the time so if the network is slow every upgrade will bring the container down for some time.

Reply:
I think for the time being we should add `--pull always` to Docker script (cc @bmanifold) and write a log message that the version is outdated on the relay/gateway and show an alert on clients when the portal detects that.